common/src/main/java/tachyon/util/network/NetworkAddressUtils.java

```diff
@@ -44,6 +45,7 @@
 /**
  * Common network address related utilities shared by all components in Tachyon.
  */
+@ThreadSafe
```

Comment:
this class needs to be thread safe if it is to be safely use by other thread safe classes (e.g. BlockStoreContext) ... I would argue that each Utils class should be thread safe, which is what people expect and when that's not the case, people end up writing racy code (e.g. gethostbyname() from the standard C library)

The issue with using a static block here is that the logic that sets `sLocalHost` and `sLocalIP` can fail. If it fails in the static block, it is never reattempted, while the current approach will try it again next time the function is requested.